EIPS/eip-2770.md

```diff
@@ -0,0 +1,178 @@
+---
+eip: 2770
+title: Meta-Transactions Forwarder Contract
+author: Dror Tirosh <dror.tirosh@gmail.com>, Alex Forshtat <forshtat1@gmail.com>
+discussions-to: https://github.com/ethereum/EIPs/pull/2770
+status: Draft
+type: Standards Track
+category: ERC
+created: 2020-07-01
+requires: 712
+---
+
+## Simple Summary
+This proposal describes a singleton Forwarder contract that enables various meta-transaction systems to co-exist without a need for recipient contracts to add support for them individually.
+
+## Abstract
+The standard way in Ethereum for smart contracts to authenticate callers is by checking the value of `msg.sender` (`CALLER 0x33`).
+For externally-owned accounts it means that the transaction signature is validated on the protocol level.
+On the other hand, this introduces a requirement for the caller account to have a sufficient amount of Ether to pay for the gas.
+In this ERC we propose to introduce a singleton contract whose responsibility is to validate transaction signatures on-chain,
+expose the signer to the destination contract and leave the gas calculation outside of the picture.
+
+## Motivation
+Ethereum is a leading platform for dapps and user acquisition is hard when having Ether is a prerequisite for even trying out the dapp.
+This lead to the emergence of the concept called ‘meta-transactions’, where signing and paying for gas is separated.
+On the other hand, there was no standard solution and a set of competing approaches emerged.
+Here we propose to establish a piece of shared infrastructure to build upon.
+
+## Specification
+The Forwarder contract operates by accepting a signed typed data together with it's ERC-712 signature,
+performs signature verification of incoming data, appends the signer address to the data field and performs a call to the target.
+
+#### Forwarder data type registration
```

Comment:
```suggestion
### Forwarder data type registration
```
Same for other subsections.  The ToC renders incorrectly in some places when you skip nesting levels.